journal/journal.go

```diff
@@ -84,31 +104,27 @@ func Send(message string, priority Priority, vars map[string]string) error {
 		appendVariable(data, k, v)
 	}
 
-	_, err := io.Copy(conn, data)
-	if err != nil && isSocketSpaceError(err) {
+	_, _, err := conn.WriteMsgUnix(data.Bytes(), nil, socketAddr)
+	if err != nil {
+		if !isSocketSpaceError(err) {
+			return err
+		}
+
+		// Large log entry, send it via tempfile and ancillary-fd.
 		file, err := tempFd()
 		if err != nil {
-			return journalError(err.Error())
+			return err
```

Comment:
nit: some error wrapping would be nice here to have context at the caller's side.